garak/payloads.py

```diff
@@ -55,11 +53,9 @@ def load_payload(
     if path is not None:
         return PayloadGroup(name, path)
     else:
-        # iterate through search dirs
-        for dir in PAYLOAD_SEARCH_DIRS:
-            path = pathlib.Path(dir) / f"{name}.json"
-            if path.is_file():
-                return PayloadGroup(name, path)
+        path = PAYLOAD_DIR / f"{name}.json"
```

Comment:
Utilizing the data_path object here will cause this to raise `GarakException` when the file is not found, as reflected in the test updates, and avoids the `FileNotFoundError` raised below.  Does it makes sense to allow the exception to bubble up or should the callers of `load_payload` be expected to handle the revised exception?

The error message on the exception will be `The resource requested does not exist {name}.json`.

I suspect the answer is that `Director.load()` should handle the `GarakException` with a log entry and raise `PayloadFailure` from that response.

Reply:
Updated in 958ea3c

Reply:
Noting that the rglob in `payloads.Director` might conflict with other assumptions about payloads being file entries in `PAYLOAD_DIR`